Fix for double border card header + list-group

gijsbotje opened the pull request:
fixes the doubled up border when a `card-header` precedes a `list-group`
fixes #21609

gijsbotje committed:
Merge pull request #1 from twbs/v4-dev

updating fork of BS to v4-beta1

gijsbotje committed:
fixes double border when .list-group.list-group-flush is preceded by .card-header

royklutman commented:
```
.card-header + .list-group > li:first-child,
.card-header + .list-group > a:first-child {
  border: 0;
 }
```

This fixes the .list-groups with links too.

gijsbotje committed:
added suggested fix for list-groups with a elements

gijsbotje committed:
Merge branch 'v4-dev' into fix/double-border-card-header

gijsbotje commented:
@royklutman added your suggestion 👍

mdo committed:
Simplify selector and only override what we set

1. Instead of having two selectors, we use the `.list-group-item` class and have coverage for list items and anchors.
2. Rather than override all border values, we only override what we set with `border-top`.

mdo commented:
Tweaked the CSS and selector just now. Demo at https://codepen.io/emdeoh/pen/KvoqxZ.
